commitlint.config.js

```diff
@@ -1,35 +1,62 @@
+const path = require('path')
+const { readdirSync } = require('fs')
+
+const appsDir = path.join(__dirname, 'apps')
+const packagesDir = path.join(__dirname, 'packages')
+
+const getAppNames = (dir) => readdirSync(dir, { withFileTypes: true })
+    .filter(file => file.isDirectory())
+    .map(file => file.name)
+
 module.exports = {
+    extends: ['@commitlint/config-conventional'],
     plugins: ['commitlint-plugin-function-rules'],
     rules: {
-        'function-rules/subject-case': [
-            2, // level: error
+        'header-max-length': [1, 'always', 52],
+        'body-max-line-length': [1, 'always', 72],
+        'type-enum': [2, 'always', [
+            'build',
+            'chore',
+            'ci',
+            'docs',
+            'feat',
+            'fix',
+            'perf',
+            'refactor',
+            'revert',
+            'style',
+            'test',
+        ]],
+        'scope-enum': [2, 'always', [
+            ...getAppNames(appsDir),
+            ...getAppNames(packagesDir),
+        ]],
+        'subject-case': [1, 'always', ['sentence-case', 'start-case', 'pascal-case', 'upper-case']],
+        'function-rules/subject-min-length': [
+            2,
             'always',
-            ({ header }) => {
-                const standalonePrefixes = ['HOTFIX', 'INFRA', 'REFACTOR']
-                for (let prefix of standalonePrefixes) {
-                    if (header.match(new RegExp(`^${prefix} \\w+`, 'i'))) {
-                        return [true]
-                    }
+            ({ subject }) => {
+                if (!subject || !subject.match(/DOMA-\d+(\s+\S+){2,}/)) {
```

Comment:
What about non task prefixes? Like INFRA, HOTFIX and etc?

Reply:
Added INFRA subject prefix, added hot fix type which does not require a task number